pkgs/by-name/li/libbgcode/package.nix

```
@@ -0,0 +1,40 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, cmake
+, pkg-config
+, heatshrink
+, zlib
+, boost
+, catch2
+}:
+stdenv.mkDerivation {
+  pname = "libbgcode";
+  version = "2023-11-16";
+
+  src = fetchFromGitHub {
+    owner = "prusa3d";
+    repo = "libbgcode";
+    rev = "bc390aab4427589a6402b4c7f65cf4d0a8f987ec";
+    hash = "sha256-TZShYeDAh+fNdmTr1Xqctji9f0vEGpNZv1ba/IY5EoY=";
+  };
+
+  nativeBuildInputs = [
+    cmake
+    pkg-config
+  ];
+
+  buildInputs = [
+    heatshrink
+    zlib
+    boost
+    catch2
+  ];
+
+  meta = with lib; {
+    homepage = "https://github.com/prusa3d/libbgcode";
+    description = "Prusa Block & Binary G-code reader / writer / converter";
+    license = licenses.agpl3;
```

```suggestion
    license = licenses.agpl3Plus;
```
According to https://github.com/prusa3d/libbgcode/blob/bc390aab4427589a6402b4c7f65cf4d0a8f987ec/pyproject.toml#L12

---

Isn't that only for python bindings, which aren't built there? There is no other explicit mentions of license, so I assumed agpl-3-only

---

OK, maybe

---

Up to you, since no clear statement from upstream.